packages/datagrid/src/components/datagrid.component.js

```diff
@@ -0,0 +1,246 @@
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+import { AgGridReact } from 'ag-grid-react';
+import 'ag-grid/dist/styles/ag-grid.css';
+import { Inject } from '@talend/react-components';
+
+import { HEADER_RENDERER_COMPONENT } from './default-header-renderer';
+import { CELL_RENDERER_COMPONENT } from './default-cell-renderer';
+import { PIN_HEADER_RENDERER_COMPONENT } from './default-pin-header-renderer';
+
+import { DATAGRID_PROPTYPES } from './datagrid.proptypes';
+import { NAMESPACE_INDEX } from './constants';
+import sampleSerializer from './sample-serializer';
+import theme from './datagrid.scss';
+
+const FOCUSED_COLUMN_CLASS_NAME = 'column-focus';
+const AG_GRID_CUSTOM_HEADER_KEY = 'headerComponent';
+const AG_GRID_CUSTOM_CELL_KEY = 'cellRenderer';
+const AG_GRID_DEFAULT_ROW_SELECTION = 'single';
+const HEADER_HEIGHT = 69;
+const ROW_HEIGHT = 39;
+
+export function injectedHeaderRenderer(getComponent, headerRenderer, onFocusedColumn) {
+	const Component = Inject.get(getComponent, headerRenderer);
+
+	return props => <Component {...props} onFocusedColumn={onFocusedColumn} />;
+}
+
+export function injectedCellRenderer(getComponent, cellRenderer, avroRenderer) {
+	const Component = Inject.get(getComponent, cellRenderer);
+
+	return props => <Component {...props} avroRenderer={avroRenderer} getComponent={getComponent} />;
+}
+
+function getAvroRenderer(avroRenderer) {
+	return {
+		booleanCellRenderer: 'DefaultBooleanCellRenderer',
+		dateCellRenderer: 'DefaultDateCellRenderer',
+		intCellRenderer: 'DefaultIntCellRenderer',
+		stringCellRenderer: 'DefaultStringCellRenderer',
+		...avroRenderer,
+	};
+}
+
+export default class DataGrid extends React.Component {
+	static defaultProps = {
+		cellRenderer: 'DefaultCellRenderer',
+		getPinnedColumnDefsFn: sampleSerializer.getPinnedColumnDefsFromSample,
+		getColumnDefsFn: sampleSerializer.getColumnDefsFromSample,
+		getRowDataFn: sampleSerializer.getRowDataFromSample,
+		getValueGetterFn: sampleSerializer.valueGetterFromRowData,
+		headerHeight: HEADER_HEIGHT,
+		headerRenderer: 'DefaultHeaderRenderer',
+		pinHeaderRenderer: 'DefaultPinHeaderRenderer',
+		rowHeight: ROW_HEIGHT,
+		rowSelection: AG_GRID_DEFAULT_ROW_SELECTION,
+	};
+
+	static propTypes = DATAGRID_PROPTYPES;
+
+	constructor(props) {
+		super(props);
+
+		this.handleKeyboard = this.handleKeyboard.bind(this);
+		this.onFocusedColumn = this.onFocusedColumn.bind(this);
+		this.onFocusedCell = this.onFocusedCell.bind(this);
+		this.onGridReady = this.onGridReady.bind(this);
+		this.setGridElement = this.setGridElement.bind(this);
+		this.setCurrentFocusedColumn = this.setCurrentFocusedColumn.bind(this);
+		this.updateStyleFocusColumn = this.updateStyleFocusColumn.bind(this);
+	}
+
+	onGridReady({ api }) {
+		this.gridAPI = api;
+	}
+
+	onFocusedCell({ column, ...rest }) {
+		if (!column) {
+			return;
+		}
+
+		if (column.colId !== this.currentColId || column.pinned) {
+			this.removeFocusColumn();
```

Comment:
**The problem**

There is an implementation issue. 
On cell focus
* we manipulate DOM to remove a className
* we save current col ref in order to fix classnames issues on scroll
* we manipulate DOM to add a className

On vertical and horizontal scroll
* we manipulate DOM to add a className

**Proposal**

Ping @acateland too since he commented about that.
Perhaps we can do that better. Here's a proposal to discuss and improve : 
* we set a convention, where cell renderers should pass down a className props to its rendered top div
* you wrap the cell, header, pinned-header component that pass a className if the column is selected, based on a `state.currentColId`
* to avoid creating a component function at each render : 
* * in constructor, you set the cell, header, pinned-header component in state.
* * you listen props (componentWillReceiveProps) to update those components when they change

Reply:
I try to develop this by another way but it's not concluding.
 - always need to manipulate the DOM
 - many constraints in the code to do that
 - bad fps when we change the focus on another cell.

The test on this plunker: https://plnkr.co/edit/eRq3ciGVSENgsQGRcjHK?p=preview

I have modified the way to access the DOM. I use the ref on AgGridReact to access to the DOM of this instance.